### ISSUE TITLE
Some sort of train wreck just happened. Videos randomly got set to private. 99% of categories do not appear.

### ISSUE THREAD

**Issue by walt93** [WWBN/AVideo#6973]
**Describe the question**
I am trying to get a clear and concise picture of what happened. I shared a link to a video I've shared a hundred times before. Immediately got feedback that the video was marked private. I went to the site, logged in, made the video public - ok - no idea how the hell that happened. 

Now I am looking at the categories in the left menu and 98% of them are gone unless you are logged in.

How are my categories "private"? and what would cause such a thing when I am literally the only person administrating the site?

**Screenshots**

Where are all of the categories?

Not logged in:
<img width="410" alt="Screen Shot 2022-08-06 at 7 40 36 PM" src="https://user-images.githubusercontent.com/11711050/183272664-d50b09de-7031-4776-aa79-7b62982a5c1e.png">

Logged in:
<img width="390" alt="Screen Shot 2022-08-06 at 7 41 21 PM" src="https://user-images.githubusercontent.com/11711050/183272671-7637ca9a-a54d-441e-96d2-7db1e5d213b3.png">

Here is a category that should be visible. It is marked "public" - even though that has nothing to do with the category being visible in the left menu I just thought I'd point that out since I have literally no idea what is going on, just that site traffic is down, videos that should be public are private, and my category menu is gone.....

<img width="2127" alt="Screen Shot 2022-08-06 at 7 42 39 PM" src="https://user-images.githubusercontent.com/11711050/183272696-a86b1b16-a6a3-413b-b61d-45e0393aeaa8.png">

**Comment by walt93** [comment]
UPDATE: Found the trigger.

I have thousands of videos in hundreds of categories. Because ROKU support is restricted to categories - I set up a separate hierarchy of categories for ROKU & made them "private" by putting group membership on them. For whatever (bug) reason, this makes most-but-not-all of the categories private.

TL;dr getting ROKU support for programs/playlists will make this nightmare easier. I am re-importing hundreds of files, just so I don't screw up the rather intricate category setup - but I cannot have that parallel hierarchy hidden without screwing up the rest of the site.....

**Comment by DanielnetoDotCom** [comment]
Hi, 

what exactly are you trying to do?

1. are you trying to have a specific category for Roku only? 
2. and this category should not be visible to your users?
3. I do not get what is wrong when you set a user group for that category

**Comment by walt93** [comment]
Daniel, yes, I was trying to set up specific categories for Roku only & hide them from the users.

When I did that, for whatever reason, all but two of my categories went "missing". I've fixed it by removing the group membership from the Roku-specific categories.

How far down the pipeline is Roku JSON from programs?
